examples/gno.land/r/demo/social_feeds/feed.gno

```diff
@@ -0,0 +1,77 @@
+package social_feeds
+
+import (
+	"std"
+	"time"
+
+	"gno.land/p/demo/avl"
+)
+
+type FeedID uint64
+
+type Feed struct {
+	id        FeedID
+	url       string
+	name      string
+	creator   std.Address
+	owner     std.Address
+	posts     avl.Tree // Post.id -> *Post
+	createdAt time.Time
+	deleted   avl.Tree
+
+	postsCtr uint64
+}
+
+func newFeed(fid FeedID, url string, name string, creator std.Address) *Feed {
+	if !reName.MatchString(name) {
+		panic("invalid feed name: " + name)
+	}
+
+	if gFeedsByName.Has(name) {
+		panic("feed already exists: " + name)
+	}
+
+	return &Feed{
+		id:        fid,
+		url:       url,
+		name:      name,
+		creator:   creator,
+		owner:     creator,
+		posts:     avl.Tree{},
+		createdAt: time.Now(),
+		deleted:   avl.Tree{},
+
+		postsCtr: 0,
+	}
+}
+
+func (feed *Feed) incGetPostID() PostID {
+	feed.postsCtr++
+	return PostID(feed.postsCtr)
+}
+
+func (feed *Feed) GetPost(pid PostID) *Post {
```

Comment:
these should return a copy of Post, otherwise other realms could arbitrarily call `Post.Update` or `Post.Delete` for example

Reply:
actually there is a mechanism where a realm cannot modify objects owned by other realms so this might not be needed

just FYU: currently there is a bug in it that allows to change ownership when you store a pointer that is passed to you